Pinta.Tools/Editable/EditEngines/BaseEditEngine.cs

```diff
@@ -457,186 +459,223 @@ public virtual void HandleAfterRedo ()
 	public virtual bool HandleKeyDown (Document document, ToolKeyEventArgs e)
 	{
 		Gdk.Key keyPressed = e.Key;
+		switch (keyPressed) {
+			case Gdk.Key.Delete:
+				HandleDelete ();
+				return true;
+			case Gdk.Key.Return:
+			case Gdk.Key.KP_Enter:
+				//Finalize every editable shape not yet finalized.
+				FinalizeAllShapes ();
+				return true;
+			case Gdk.Key.space:
+				HandleSpace (e);
+				return true;
+			case Gdk.Key.Up:
+				//Make sure a control point is selected.
+				HandleUp ();
+				return true;
+			case Gdk.Key.Down:
+				//Make sure a control point is selected.
+				HandleDown ();
+				return true;
+			case Gdk.Key.Left:
+				//Make sure a control point is selected.
+				HandleLeft (e);
+				return true;
+			case Gdk.Key.Right:
+				//Make sure a control point is selected.
+				HandleRight (e);
+				return true;
+			default:
+				if (keyPressed.IsControlKey ()) {
+					// Redraw since the Ctrl key affects the hover cursor, etc
+					HandleControls (e);
```

Comment:
This is just a one-liner so I would just inline the `DrawActiveShape` call instead of having a separate method

Reply:
No problem, I just made the changes you are requesting